app/js/components/pages/Job.jsx

```diff
@@ -8,7 +8,7 @@ const Job = ({ plugin, workflow, inputArtifacts, submitJob, cancelJob }) => {
             <div className="page-header">
                 <h1>{plugin.name}: {workflow.description}</h1>
             </div>
-            <form>
+            <form onSubmit={(e) => { e.preventDefault(); submitJob(workflow); }}>
```

Comment:
I think you might be able to pass `e` (which is a `this` object bound to the DOM node) to `submitJob` making it context sensitive instead of finding a "form" and running with it).